src/app/components/pool/graphs/heatmap/nodes-heatmap.component.ts

```diff
@@ -212,23 +217,24 @@ export class NodesHeatmapComponent implements AfterViewInit, OnChanges, OnDestro
         this.redraw();
     }
 
-    private _updateSvg(groups: any) {
+    private _updateSvg(groups: NodeSelection) {
         const spaceInBetweenNode = Math.ceil(this.dimensions.tileSize / 20);
         const z = Math.max(this.dimensions.tileSize - spaceInBetweenNode, 0);
         const nodeEnter = groups.enter().append("g")
             .attr("class", "node-group")
-            .on("click", (tile) => {
+            .on("click", (_event: MouseEvent, tile: HeatmapTile) => {
                 if (!this.interactive) {
                     return;
                 }
                 this.selectedNodeId.next(tile.node.id);
-                this._updateSvg(this._svg.selectAll("g.node-group"));
-            }).on("dblclick", (tile) => {
+                this._updateSvg(this._svg
+                    .selectAll<SVGGElement, HeatmapTile>("g.node-group"));
```

Comment:
I don't think you need to specify the generic types here (ie: it looks like `this._updateSvg(this._svg.selectAll("g.node-group"));` should work)